cmd/sonobuoy/app/args.go

```diff
@@ -468,3 +431,39 @@ func (m *Mode) Set(str string) error {
 	m.name = lcase
 	return nil
 }
+
+type addConfigMapFlag struct {
```

Comment:
Right now the set command is extremely tied to specifically the e2e custom repo config since it also adds the env var.

I think it could be generalized some but that can definitely be done in a follow up PR if we generalize the logic even more (only if we want), for instance to add configmaps to plugins on the fly. I don't think that feature is typically needed though and `sonobuoy gen plugin` can also be used as needed or direct yaml manipulation.